From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Paul Praschl <praschlpaul@g-p.at>
Date: Sun, 2 Aug 2026 23:01:43 +0200
Subject: [PATCH 091/117] arm64: dts: apple: t8010: Add UART1

Bluetooth on T8010 boards uses UART1. Describe the controller and
publish its serial alias. Leave it disabled until a board selects it.

Signed-off-by: Paul Praschl <praschlpaul@g-p.at>

Orchard-replay-of: 363034f23461f23c67684f805d17a45c10a0321b
---
 arch/arm64/boot/dts/apple/t8010-common.dtsi |  1 +
 arch/arm64/boot/dts/apple/t8010.dtsi        | 13 +++++++++++++
 2 files changed, 14 insertions(+)

diff --git a/arch/arm64/boot/dts/apple/t8010-common.dtsi b/arch/arm64/boot/dts/apple/t8010-common.dtsi
index 23938137d..92df55da6 100644
--- a/arch/arm64/boot/dts/apple/t8010-common.dtsi
+++ b/arch/arm64/boot/dts/apple/t8010-common.dtsi
@@ -14,6 +14,7 @@
 		ausbphy0 = &ausbphy0; /* Used by loader */
 		nvme-hmb = &nvme_hmb; /* Used by loader */
 		serial0 = &serial0;
+		serial1 = &serial1;
 		serial3 = &serial3;
 		serial5 = &serial5;
 	};
diff --git a/arch/arm64/boot/dts/apple/t8010.dtsi b/arch/arm64/boot/dts/apple/t8010.dtsi
index 4ead85563..8dc02eb60 100644
--- a/arch/arm64/boot/dts/apple/t8010.dtsi
+++ b/arch/arm64/boot/dts/apple/t8010.dtsi
@@ -165,6 +165,19 @@
 			status = "disabled";
 		};
 
+		serial1: serial@20a0c4000 {
+			compatible = "apple,t8010-uart", "apple,s5l-uart";
+			reg = <0x2 0x0a0c4000 0x0 0x4000>;
+			reg-io-width = <4>;
+			interrupt-parent = <&aic>;
+			interrupts = <AIC_IRQ 219 IRQ_TYPE_LEVEL_HIGH>;
+			/* Use the bootloader-enabled clocks for now. */
+			clocks = <&clkref>, <&clkref>;
+			clock-names = "uart", "clk_uart_baud0";
+			power-domains = <&ps_uart1>;
+			status = "disabled";
+		};
+
 		serial3: serial@20a0cc000 {
 			compatible = "apple,t8010-uart", "apple,s5l-uart";
 			reg = <0x2 0x0a0cc000 0x0 0x4000>;
